From 6087c28987e2e24a574ea0b712986553dd02f20a Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Wed, 15 Mar 2017 17:48:28 +0000
Subject: [PATCH] RenderManager: Force frames to output when reconfiguring
 renderer

Currently the clock may be paused before the final queued frame is rendered.
This prevents the queue of pictures to render from being drained and
so stops the PRESENT_IDLE state being reached.

This results in a 5 seconds pause and the following warning:
"CRenderManager::Configure - timeout waiting for state"

This has been observed on Pi in DVD menus.
---
 xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp
index 5230c6d..9b1203f 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp
@@ -212,15 +212,18 @@ bool CRenderManager::Configure(DVDVideoPicture& picture, float fps, unsigned fla
   {
     CSingleLock lock(m_presentlock);
     XbmcThreads::EndTime endtime(5000);
+    m_forceNext = true;
     while (m_presentstep != PRESENT_IDLE)
     {
       if(endtime.IsTimePast())
       {
         CLog::Log(LOGWARNING, "CRenderManager::Configure - timeout waiting for state");
+        m_forceNext = false;
         return false;
       }
       m_presentevent.wait(lock, endtime.MillisLeft());
     }
+    m_forceNext = false;
   }
 
   {
-- 
1.9.1

